File: python_translation/entry_points/H5V1Refactor.py
"""
H5 V1 Refactor entry point (translated from Java)
Original: com/idlmlpugdw/h5_v1_refactor/H5V1Refactor.java
"""

from __future__ import annotations
from typing import Optional, Any
import threading
import json
import sys
import os

# Add parent directory to path
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

# Android placeholder types
Context = Any  # Android Context placeholder

try:
    from models import Atom
except ImportError:
    Atom = Any


class H5V1Refactor:
    """
    H5 V1 Refactor entry point class.
    Initializes the H5 system with JSON configuration and Android context.

    Original package: com.idlmlpugdw.h5_v1_refactor
    """

    class Worker:
        """
        Inner worker class that runs initialization in a separate thread.
        Original inner class: llllIIIIll1
        """

        def __init__(self, json_config: str, context: Context):
            """
            Initialize worker.

            Args:
                json_config: JSON configuration string
                context: Android context
            """
            self.json_config = json_config
            self.context = context

        def run(self):
            """
            Run the initialization task.
            Parses JSON config and initializes the system.
            """
            try:
                # Import at runtime to avoid circular dependencies
                from utils.IlIlllIIlI1.llllIllIl1 import llllIllIl1
                from utils.lIIlllIIIlllII1.llllIllIl1 import llllIllIl1 as ConfigClass

                # Parse JSON configuration
                config_obj = json.loads(self.json_config)

                # Create Atom from JSON
                if Atom is not Any:
                    atom = Atom.from_json_object(config_obj)
                else:
                    atom = config_obj

                # Initialize the system
                # llllIllIl1().llllIIIIll1(context, session_id, atom)
                initializer = llllIllIl1()
                session_id = getattr(ConfigClass, 'f377llllIllIl1', 'default_session')

                if hasattr(initializer, 'llllIIIIll1'):
                    initializer.llllIIIIll1(self.context, session_id, atom)

            except json.JSONDecodeError as e:
                raise RuntimeError(f"JSON parsing error: {e}")
            except Exception as e:
                raise RuntimeError(f"Initialization error: {e}")

    @staticmethod
    def init(context: Context, json_config: str):
        """
        Initialize H5 system with configuration.

        Args:
            context: Android context
            json_config: JSON configuration string

        Example:
            H5V1Refactor.init(android_context, '{"key": "value"}')
        """
        worker = H5V1Refactor.Worker(json_config, context)
        thread = threading.Thread(target=worker.run)
        thread.start()


def init(context: Context, json_config: str):
    """
    Convenience function for H5 initialization.

    Args:
        context: Android context
        json_config: JSON configuration string
    """
    H5V1Refactor.init(context, json_config)

File: sources/com/idlmlpugdw/h5_v1_refactor/H5V1Refactor.java
package com.idlmlpugdw.h5_v1_refactor;

import IlIlllIIlI1.llllIllIl1;
import android.content.Context;
import c13.nim5.ez8.h5_proto.Atom;
import org.json.JSONException;
import org.json.JSONObject;

/* loaded from: classes.jar:com/idlmlpugdw/h5_v1_refactor/H5V1Refactor.class */
public class H5V1Refactor {

    /* loaded from: classes.jar:com/idlmlpugdw/h5_v1_refactor/H5V1Refactor$llllIIIIll1.class */
    public class llllIIIIll1 implements Runnable {

        /* renamed from: llllIIIIll1, reason: collision with root package name */
        public final /* synthetic */ String f326llllIIIIll1;

        /* renamed from: lIIIIlllllIlll1, reason: collision with root package name */
        public final /* synthetic */ Context f327lIIIIlllllIlll1;

        public llllIIIIll1(String str, Context context) {
            this.f326llllIIIIll1 = str;
            this.f327lIIIIlllllIlll1 = context;
        }

        @Override // java.lang.Runnable
        public void run() {
            try {
                new llllIllIl1().llllIIIIll1(this.f327lIIIIlllllIlll1, lIIlllIIIlllII1.llllIllIl1.f377llllIllIl1, Atom.fromJSONObject(new JSONObject(this.f326llllIIIIll1)));
            } catch (JSONException e) {
                throw new RuntimeException(e);
            }
        }
    }

    public static void init(Context context, String str) {
        new Thread(new llllIIIIll1(str, context)).start();
    }
}
